### PR TITLE
add a new approach for IsPalindrom method

### PR DESCRIPTION
**Pull request by HaoZhang615**

### CODE REVIEW

File: IsPalindrom/Program.cs [hunk 1]
```diff
@@ -15,4 +12,19 @@ bool isPalindrom (string stringToTest)
...
-System.Console.WriteLine(isPalindrom("racecar"));
\ No newline at end of file
+System.Console.WriteLine(isPalindrom("racecar"));
+
+// approach 2 with Array.Reverse method
+bool isPalindrom2 (string stringToTest)
+{
+    char [] tempCharArray = stringToTest.ToCharArray();
+    // Array.Reverse is a method with void type, means doesn't return anything
+    Array.Reverse(tempCharArray);
```
Comment (mfdanb) [inline IsPalindrom/Program.cs:22]: this is a good solution, congrats!

we can make it shorter though:

strings are characters arrays technically, so you dont really need the ToCharArray part if you use them like this.

and you can use Array.Reverse() as an extension method on strings as well, further shortening the code. (https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods)

resulting in something like this:

return stringToTest == stringToTest.Reverse();


LINQ (https://docs.microsoft.com/en-us/dotnet/standard/linq/) also has a SequenceEqual method, to check if two collections/sequences are equal. for custom types you would have to implement your own equality comparer method, but for base types (like string) it works out of the box.

so your entire method can be written like this:

return stringToTest.SequenceEqual(stringToTest.Reverse());

File: IsPalindrom/Program.cs [hunk 2]
```diff
@@ -1,7 +1,4 @@
-﻿//Step 1 - create a new empty string "container"
-//step 2 - read the input string "backwards" and load the characters one by one into the new empty string "container"
-//step 3 - compare if the new string equals to the input string
-
+﻿// approach 1 with for loop
 bool isPalindrom (string stringToTest)
```
Comment (mfdanb) [inline IsPalindrom/Program.cs:2]: for the first isPalindrom method:

also good!

the only thing that comes to mind is that you are repeatedly allocating a new array in your for loop:

```

bool isPalindrom (string stringToTest) 
{ 
    string stringReversed = "";

    for (int i = stringToTest.ToCharArray().Length-1; i >= 0; i--)  <------- .ToCharArray once
    { 
        stringReversed += stringToTest.ToCharArray()[i]; <------ .ToCharArray i times
    }

    return stringToTest == stringReversed;
  
}
```

thats a lot of allocations. (i+1 times)

if you gonna use the char array this many time, I'd recommend you create a variable for it, and use that everywhere

File: IsPalindrom/Program.cs (hunk 1 above)
Comment (HaoZhang615) [reply IsPalindrom/Program.cs:22]: man this is some nice and simple implementation... thank you!!

File: IsPalindrom/Program.cs (hunk 2 above)
Comment (HaoZhang615) [reply IsPalindrom/Program.cs:2]: very good advise! This forces me to think about how the computer is gonna run the method in real world and to improve performance.

**Pull request merged**